cupy/random/generator.py

```diff
@@ -235,6 +235,74 @@ def uniform(self, low=0.0, high=1.0, size=None, dtype=float):
         rand = self.random_sample(size=size, dtype=dtype)
         return dtype.type(low) + rand * dtype.type(high - low)
 
+    def choice(self, a, size=None, replace=True, p=None):
+        """Returns an array of random values from a given 1-D array.
+
+        .. seealso::
+            :func:`cupy.random.choice` for full document,
+            :meth:`numpy.random.choice`
+
+        """
+        if a is None:
+            raise ValueError('a must be 1-dimensional or an integer')
+        if isinstance(a, six.integer_types):
+            a_size = a
+            if a_size <= 0:
+                raise ValueError('a must be greater than 0')
+            a = cupy.array(a, copy=False)
+        else:
+            a = cupy.array(a, copy=False)
+            if a.ndim != 1:
+                raise ValueError('a must be 1-dimensional')
+            elif a.dtype != 'int':
```

Comment:
It seems to me that when `a` is array_like, its dtype need not to be integer. Actually, a is an array of strings in some example of `NumPy`.